corehq/motech/generic_inbound/models.py

```diff
@@ -72,3 +76,50 @@ def to_json(self):
             "expression_id": self.expression_id,
             "message": self.message,
         }
+
+
+class RequestLog(models.Model):
+
+    class Status(models.TextChoices):
+        FILTERED = 'filtered', _('Filtered')
+        VALIDATION_FAILED = 'validation_failed', _('Validation Failed')
+        SUCCESS = 'success', _('Success')
+        ERROR = 'error', _('Error')
+        REVERTED = 'reverted', _('Reverted')
+
+    class RequestMethod(models.TextChoices):
+        POST = 'POST'
+        PUT = 'PUT'
+        PATCH = 'PATCH'
+
+    id = models.UUIDField(primary_key=True, default=uuid4)
+    domain = models.CharField(max_length=255, db_index=True)
+    api = models.ForeignKey(ConfigurableAPI, on_delete=models.CASCADE)
+    status = models.CharField(max_length=32, choices=Status.choices, db_index=True)
+
+    timestamp = models.DateTimeField(auto_now=True, db_index=True)
+    attempts = models.PositiveSmallIntegerField(default=1)
+    response_status = models.PositiveSmallIntegerField(db_index=True)
+    error_message = models.TextField()
+
+    username = models.CharField(max_length=128, db_index=True)
+    request_method = models.CharField(max_length=32, choices=RequestMethod.choices)
+    request_query = models.CharField(max_length=8192)
+    request_body = models.TextField()
+    request_headers = models.JSONField(default=dict)
+    request_ip = models.GenericIPAddressField(db_index=True)
+
+
+class ProcessingAttempt(models.Model):
+    log = models.ForeignKey(RequestLog, on_delete=models.CASCADE)
+    timestamp = models.DateTimeField(auto_now=True, db_index=True)
+    is_retry = models.BooleanField(default=False)
+    response_status = models.PositiveSmallIntegerField(db_index=True)
+    response_body = models.TextField()
+    raw_response = models.JSONField(default=dict)
+
+    xform_id = models.UUIDField(db_index=True, null=True, blank=True)
+    case_ids = ArrayField(models.UUIDField(default=uuid4), null=True, blank=True)
```

Comment:
I'm thinking about switching these from `UUIDField` to `CharField`.  I like the idea of using UUIDs, and Django does normalize the form, which is pretty neat, but I think it's perhaps too far outside the norm of how we treat IDs in HQ.

Reply:
@snopoke do you mean that you agree I should switch this back to `CharField`?

Reply:
I think so, yes